python/dolfinx/wrappers/mesh.cpp

```diff
@@ -457,10 +457,28 @@ void mesh(nb::module_& m)
       "build_dual_graph",
       [](const MPICommWrapper comm, dolfinx::mesh::CellType cell_type,
          const dolfinx::graph::AdjacencyList<std::int64_t>& cells)
-      { return dolfinx::mesh::build_dual_graph(comm.get(), cell_type, cells); },
+      {
+        std::vector<dolfinx::mesh::CellType> c = {cell_type};
+        return dolfinx::mesh::build_dual_graph(comm.get(), std::span{c},
+                                               {cells.array()});
+      },
       nb::arg("comm"), nb::arg("cell_type"), nb::arg("cells"),
       "Build dual graph for cells");
 
+  m.def(
+      "build_dual_graph",
+      [](const MPICommWrapper comm,
+         std::vector<dolfinx::mesh::CellType>& cell_types,
+         const std::vector<std::vector<std::int64_t>>& cells)
```

Comment:
This is a nasty argument type for Python/nanobind. Probably should be a `std::vector` of  `nb::ndarray` objects.

Reply:
OK, something like that. I've updated it.